### ISSUE TITLE
zones with bignum seems to be imprecise for computing summaries

### ISSUE THREAD

**Issue by caballa** [seahorn/crab#19]
The tests 
- `simple/test-arr-4.c` 
- `simple/test-arr-4.unsafe.c` 
- `simple/fntest-1.c`
- `simple/fntest-1.unsafe.c`

from crab-llvm do not pass if we use:

```
  typedef SDBM_impl::BigNumDefaultParams<number_t> SplitDBMGraph;
  typedef SplitDBM<number_t, varname_t, SplitDBMGraph> BASE(split_dbm_domain_t);
```

in `include/crab_llvm/crab_domains.hh`

**Comment by caballa** [comment]
This is probably a symptom of a much bigger issue. 

The DBM implementation assumes that the graph weights are basic types. The assumption for basic types is because we use memcpy and realloc for efficiency reasons.  Thus, to use bignums as weights we should change the implementation which might affect performance. For instance,
 
- `domains/graphs/adapt_sgraph.hpp`
- `domains/graphs/util/Vec.h`

This is related to issue #21. Thanks to @elazarg  to bring it in!

**Comment by caballa** [comment]
Fixed in commit 85330f0cf4e336087e0d969e31da7b6135160f7b
